engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/TableSource.scala

```diff
@@ -87,11 +87,26 @@ class TableSource(
   override def testRecordParser: TestRecordParser[RECORD] = (testRecords: List[TestRecord]) =>
     FlinkMiniClusterTableOperations.parseTestRecords(testRecords, tableDefinition.toFlinkSchema)
 
-  override def generateTestData(size: Int): TestData =
-    FlinkMiniClusterTableOperations.generateTestData(size, tableDefinition.toFlinkSchema)
+  override def generateTestData(size: Int): TestData = {
+    val mode =
+      if (useRealDataForTests) DataGenerationMode.RealDataMode(sqlStatements, tableDefinition.tableName)
+      else DataGenerationMode.RandomDataMode
+    FlinkMiniClusterTableOperations.generateTestData(size, tableDefinition.toFlinkSchema, mode)
+  }
+
 }
 
 object TableSource {
   type RECORD = java.util.Map[String, Any]
   private val filteringInternalViewName = "filteringView"
+
+  private[source] def executeSqlAndGetTable(
```

Comment:
I would split it into two methods. Also, I see that "execute SQL" / "SQL statements" are too much generic names and it can cause confusions. I think that we need at least to add `DDL` word to make it clear.